src/routers/incompleteApplication.js

```diff
@@ -0,0 +1,45 @@
+const express = require('express')
+const IncompleteApplication = require('../models/IncompleteApplicationModel')
+const router = new express.Router()
+const authMiddleware = require('../middleware/authMiddleware')
+
+router.put('/editing', authMiddleware, async (req, res) => {
+  console.log(req.body)
+  try {
+    const existingIncompleteApp = await IncompleteApplication.findOne({ owner: req.user._id })
+    if (existingIncompleteApp) {
+      const updates = Object.keys(req.body)
+      const allowedUpdates = ['name', 'email', 'phoneNumber', 'university', 'major', 'classYear', 'shirtSize', 'ethnicity', 'gender', 'pronouns', 'github', 'linkedin', 'personalSite', 'otherSite', 'originCity', 'campusParking', 'travelSponsorship', 'validAge', 'dietaryRestrictions', 'allergies', 'referral', 'personalProject', 'hackathonGoal', 'mlhSignature']
+      const isValidOperation = updates.every((update) => allowedUpdates.includes(update))
+      if (!isValidOperation) {
+        return res.status(400).send({ error: 'Invalid updates' })
+      }
+      for (var update of updates) {
+        existingIncompleteApp[update] = req.body[update]
+      }
+      await existingIncompleteApp.save()
+      return res.status(201).send(existingIncompleteApp)
+    }
+    const incompleteApp = new IncompleteApplication(req.body)
+    incompleteApp.owner = req.user._id
+    await incompleteApp.save()
+    res.status(200).send(incompleteApp)
+  } catch (err) {
+    res.status(500).send({ errorMessage: err.message })
+  }
+})
+
+router.get('/editing', authMiddleware, async (req, res) => {
```

Comment:
Is this suppose to be a get request? In addition I don't think '/editing' is a sufficient route name.
Consider maybe something like /incomplete-application/editing. By convention a GET http request  should probably be something like /incomplete-applications/:id since in this case you are pulling a single incomplete application. Maybe reference your code for application routes since this is so similar.

Reply:
The '/editing' name was something I named.

I see why it is not very concise. Perhaps for the name we can do '/in-progress' or '/saves'?

The GET request should be identical so the GET request on the /applications route